logstash-core/spec/logstash/agent_spec.rb

```diff
@@ -150,10 +150,10 @@
         context "with a config that does not contain reload incompatible plugins" do
           let(:second_pipeline_config) { "input { generator { } } filter { } output { }" }
 
-          it "does not reload if new config contains reload incompatible plugins" do
+          it "does upgrade the new config" do
             t = Thread.new { subject.execute }
             sleep 0.01 until subject.running_pipelines? && subject.pipelines.values.first.ready?
-            expect(subject).to receive(:upgrade_pipeline).at_least(2).times
+            expect(subject).to receive(:upgrade_pipeline).once.and_call_original
```

Comment:
I wonder why this change of expectations here?, do you remember why you initially started with `at_least(2)` and now require only once? do you have a reason behind?